### DISCUSSION TITLE
Override application.properties used in Docker

### DISCUSSION THREAD

**Discussion by rapetum228** [apache/camel-karavan#1089] [Q&A]
Hello, I am interested in how I can override the application.properties file. I run integrations through Docker,  in Portainer. In Portainer it is possible to create a service based on an image and override some files of this image in the container. I would like to know if it is possible to do the same for Camel Karavan images? 
Other options of overriding are also welcome (for example, creating several application.properties for different environments). 
In your repository you have such a folder https://github.com/apache/camel-karavan/tree/main/karavan-demo/placeholder, which has two files application.properties and local.properties. What is the correct way to use it? Help me, please.🙏🏻

**Comment by mgubaidullin** [comment]
You could use environment properties placeholders ex: `${env:APP_HOME}` and set env variables for container

**Reply by rapetum228** [reply]
Thank you. Indeed, variables defined in application.properties in Docker are interpreted as ENVs in a special way.
For example, let's write 2 variables in application.properties:

```
myrabbit.host = localhost
test.variable = 1
```
Their equivalent is in Docker ENVs, which can be overridden: MYRABBIT_HOST, TEST_VARIABLE
